### DISCUSSION TITLE
How to measure load and activation time?

### DISCUSSION THREAD

**Discussion by PEZ** [microsoft/vscode-discussions#454] [Extension Development QnA] [answered]
When trying to make extensions quick to load and activate, I'd like to conveniently be able to benchmark it. As a part of this I'm using a script that measures and compares the bundle size and the time it takes to require it.

```
% script/bundle-compare.sh /Users/pez/.vscode/extensions/betterthantomorrow.joyride-0.0.32 /Users/pez/Downloads/joyride-0.0.33-rewrite-clj-77ebb270/extension 100
Comparison of:
  1. /Users/pez/.vscode/extensions/betterthantomorrow.joyride-0.0.32
     Size: 1351978 bytes
     Load time average: 48.33 ms
  2. /Users/pez/Downloads/joyride-0.0.33-rewrite-clj-77ebb270/extension
     Size: 1591118 bytes
     Load time average: 55.13 ms
Loaded the bundles 100 times each.
Size difference: 239140  (17.6%)
Load time difference: 6.806672046184534  (14.0%)
```

That's from this PR for anyone who wants to use the scripts as a start for your benchmarking:
* https://github.com/BetterThanTomorrow/joyride/pull/152

In the same PR I added timing printouts from the start and finish of the `activate()` function. It's 2-3 ms on my machine.

This brings me to wonder what figure it is that VS Code displays on an activated extension?

<img width="463" alt="image" src="https://user-images.githubusercontent.com/30010/218318322-80d4db86-80c4-421a-b331-fd47d85756a7.png">

It's neither the load/require time of the bundle (too low for that), nor the `activate()` function (too high).

Generally, I'd like to know what you other do in this area, what APIs/utilities are available, and if the VS Code team are interested in creating utilities/services to help with measuring and tuning our extensions startup performance.

**Comment by bwateratmsft** [comment, marked as answer]
I don't know for sure but I think what it's displaying is load time + activation time + (maybe activate resolved time?). The only way I know of that _used_ to work was to go to the running extensions page, and do "Report Issue" on one. It would put the three separate times into your clipboard. Unfortunately that doesn't work anymore.

**Reply by bwateratmsft** [reply]
I created this issue because I miss this capability: https://github.com/microsoft/vscode/issues/174255
